### ISSUE TITLE
feature request: cdklocal to use esbuild for local builds

### ISSUE THREAD

**Issue by dreamorosi** [localstack/localstack#11351]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Feature description

The `NodejsFunction` construct from AWS CDK allows you to build and deploy TypeScript functions.

Example:

```ts
import { Stack, type StackProps, CfnOutput, RemovalPolicy } from 'aws-cdk-lib';
import type { Construct } from 'constructs';
import { Runtime, Tracing } from 'aws-cdk-lib/aws-lambda';
import { NodejsFunction, OutputFormat } from 'aws-cdk-lib/aws-lambda-nodejs';
import { LogGroup, RetentionDays } from 'aws-cdk-lib/aws-logs';

export class LogeventStack extends Stack {
  constructor(scope: Construct, id: string, props?: StackProps) {
    super(scope, id, props);

    const fnName = 'LogeventFn';
    const logGroup = new LogGroup(this, 'MyLogGroup', {
      logGroupName: `/aws/lambda/${fnName}`,
      removalPolicy: RemovalPolicy.DESTROY,
      retention: RetentionDays.ONE_DAY,
    });
    const fn = new NodejsFunction(this, 'MyFunction', {
      functionName: fnName,
      logGroup,
      runtime: Runtime.NODEJS_20_X,
      entry: './src/index.ts',
      handler: 'handler',
      tracing: Tracing.ACTIVE,
      bundling: {
        minify: true,
        mainFields: ['module', 'main'],
        sourceMap: true,
        format: OutputFormat.ESM,
        banner:
          "import { createRequire } from 'module';const require = createRequire(import.meta.url);",
      },
    });

    new CfnOutput(this, 'FunctionArn', {
      value: fn.functionArn,
    });
  }
}
```

The construct relies on `esbuild` to transform and bundle your code so that it can be deployed on AWS Lambda. By default the construct will attempt to use Docker to bundle your code, however when `esbuild` is installed in the project it will perform these actions directly on the host - which is much faster and for most JavaScript cases just fine.

When working with LocalStack, even with `esbuild` installed in my project, LocalStack always uses Docker.

Ideally, I would like `cdklocal` to behave the same as when using `cdk` directly in this regard.

How to reproduce in CDK:
```sh
mkdir /tmp/test; cd /tmp/test
cdk init sample-app --language=typescript
npm i -D esbuild
```
then copy the stack above, and create a sample function.

finally, deploy with `cdk deploy` and observe that docker is never used.

### Anything else?

See here for CDK docs that explain the behavior I described: https://github.com/aws/aws-cdk/tree/main/packages/aws-cdk-lib/aws-lambda-nodejs#local-bundling

**Comment by kaykhan** [comment]
Did you manage to resolve this issue? when trying to bundle prisma i am getting `The "path" argument must be of type string. Received undefined`

**Comment by dreamorosi** [comment]
Not stale, just unattended by maintainers.

**Comment by dreamorosi** [comment]
Still not stale, still unattended by maintainers

**Comment by filipeestacio** [comment]
Yeah, I'm getting the same blocker. I find using NodejsFunction to be much cleaner in the codebase and would be my preference. If Localstack (or cdklocal) can't handle it, that's not ideal.

**Comment by HarshCasper** [comment]
Due to the forthcoming changes in our distribution and development model ([read more](https://blog.localstack.cloud/the-road-ahead-for-localstack/)), this issue tracker is no longer being maintained. We've migrated this issue to our internal tracker and will continue to work on it there. If you'd like to follow up on the status or expedite this, feel free to reach out through one of our support channels:

-   💬 [GitHub Discussions](https://github.com/orgs/localstack/discussions)
-   📖 [Help & Support](https://docs.localstack.cloud/aws/getting-started/help-support/)
